contracts/SnapshotToken/Extensions/SimpleVote.sol

```diff
@@ -0,0 +1,191 @@
+    ///   throws if proposal does not exist or the vote on it has ended already. Votes are final,
+    ///   changing the vote is not allowed
+    /// @param proposalId of the proposal to be voted on
+    /// @param voteInFavor of the desired proposal
+    function vote(uint256 proposalId, bool voteInFavor) public {
+        incrementVote(proposalId, voteInFavor, msg.sender);
+    }
+
+    /// @dev same as vote, only for a relayed vote. Will throw if provided signature (v,r,s) does not match
+    ///  the address of the voter
+    /// @param voter address whose token balance should be used as voting weight
+    function relayedVote(
+        uint256 proposalId,
+        bool voteInFavor,
+        address voter,
+        bytes32 r,
+        bytes32 s,
+        uint8 v
+    )
+        public
+    {
+        // check that message signature matches the voter address
+        // solium-disable indentation
+        require(ecrecover(
+            keccak256(abi.encodePacked(
+                "\x19Ethereum Signed Message:\n32",
+                keccak256(abi.encodePacked(byte(0), this, proposalId, voteInFavor)))),
+            v, r, s) == voter,
+        "Incorrect order signature");
+        // solium-enable indentation
+
+        incrementVote(proposalId, voteInFavor, voter);
+    }
+
+    function batchRelayedVotes(
+        uint256[] proposalIndices,
+        bool[] votePreferences,
+        address[] voters,
+        bytes32[] r,
+        bytes32[] s,
+        uint8[] v
+    )
+        public
+    {
+        require(
+            proposalIndices.length == votePreferences.length &&
+            votePreferences.length == voters.length &&
+            voters.length == r.length &&
+            r.length == s.length &&
+            s.length == v.length,
+            "Invalid voting arguments"
+        );
+        for (uint i = 0; i < voters.length; i++) {
+            relayedVote(
+                proposalIndices[i],
+                votePreferences[i],
+                voters[i],
+                r[i],
+                s[i],
+                v[i]
+            );
+        }
+    }
+
+    /// @return whether the vote on a proposal has passed, if false this could be because it was rejected or
+    ///   because the quorum was not reached.
+    /// @dev throws if vote is still ongoing or proposal does not exist. Also deletes the proposalData to free up gas.
+    //     Therefore subsequent attempts to call this function will throw too
+    function getResult(uint256 proposalId) public {
+        require(proposalId < _nextProposalId, "Invalid proposal ID");
+        Proposal storage p = _proposals[proposalId];
+        require(p.owner != address(0), "Proposal was already deleted");
+        require(p.endTime < now, "Voting period is not over");
+
+        uint256 minParticipation = TOKEN.totalSupplyAt(p.snapshotId) * 40 / 100;
+        bool passedQuorum = p.inFavor + p.against > minParticipation;
```

Comment:
non inclusive range is OK here - thanks for explanations!
practically in most governance systems (at least for companies) quorums are inclusive